packages/yarnpkg-pnpify/sources/buildNodeModulesTree.ts

```diff
@@ -259,7 +259,7 @@ const buildPackageTree = (pnp: PnpApi, options: NodeModulesTreeOptions): { packa
       if (workspace) {
         node.peerNames = new Set([
           ...pkg.packagePeers,
-          ...Array.from(workspace.manifest.peerDependencies.values()).map(x => x.scope ? `@${x.scope}/${x.name}` : x.name),
+          ...Array.from(workspace.manifest.peerDependencies.values()).map(x => structUtils.stringifyIdent(x)),
```

Comment:
```suggestion
          ...Array.from(workspace.manifest.peerDependencies.values(), x => structUtils.stringifyIdent(x)),
```
Haven't measured the performance impact here but this should™ be faster as it doesn't have to iterate the values twice

Reply:
Thank you!